### PR TITLE
Improvement/upgrade bblfshd client

### PR DESCRIPTION
**Pull request by jfontan**
Upgrade bblfsh client and add support to select mode when using `uast`.

```
uast(blob, "python", "//Identifier", "semantic")
```

### CODE REVIEW

**Comment by jfontan** [comment]
@bzz updated bblfsh minimum version in the documentation

File: internal/function/uast_test.go [hunk 1]
```diff
@@ -97,6 +113,7 @@ func TestUASTXPath(t *testing.T) {
 		{"left is nil", sql.NewRow(nil, "foo"), nil, nil},
 		{"right is nil", sql.NewRow(uast, nil), nil, nil},
 		{"both given", sql.NewRow(uast, testXPath), filteredNodes, nil},
+		// {"blob is invalid", fn4, sql.NewRow([]byte(testCode), "Python", testXPath, "invalid"), nil},
```
Comment (mcarmonaa) [inline internal/function/uast_test.go]: leftover?

File: internal/function/uast.go [hunk 2]
```diff
@@ -184,6 +202,39 @@ func (f UAST) Eval(ctx *sql.Context, row sql.Row) (out interface{}, err error) {
 		xpath = x.(string)
 	}
 
+	modeSet := false
```
Comment (ajnavarro) [inline internal/function/uast.go]: maybe is worth it to move this to a private function to improve readability?

**Comment by erizocosmico** [comment]
What happens if you want semantic uast but not want to pass language or xpath query? Being the last parameter forces you to give the other two.

**Comment by ajnavarro** [comment]
@erizocosmico you're right. Maybe we will need another UDF for semantic UAST (maybe `SUAST`? :D)

File: internal/function/uast.go [hunk 3]
```diff
@@ -40,10 +42,15 @@ func NewUAST(args ...sql.Expression) (sql.Expression, error) {
 		blob = args[0]
 		lang = args[1]
 		xpath = args[2]
+	case 4:
```
Comment (kuba--) [inline internal/function/uast.go]: I know it's a silly detail, but how about:
```
switch len(args) {
       default: return nil, sql.ErrInvalidArgumentNumber.New("1, 2, 3 or 4", len(args))
 	case 4:		 	
 		mode = args[3]
                fallthrough
 	case 3:		 	
 		xpath = args[2]
                fallthrough
 	case 2:		 	 		
 		lang = args[1]	
                fallthrough
 	case 1:
		blob = args[0]
}
```
less copy paste - up to you ;)

File: internal/function/uast.go (hunk 2 above)
Comment (jfontan) [reply internal/function/uast.go]: Is not used anymore as it is not an optional parameter.

File: internal/function/uast.go (hunk 3 above)
Comment (jfontan) [reply internal/function/uast.go]: I've left it was it was previously as `uast` now changed back to its original parameters. We can do a second round of refinement if we feel it's needed.

**Comment by jfontan** [comment]
Implemented `uast_mode(blob, language, mode)` and `uast` now has the same parameters as before but returns `semantic` version of AST.

**Comment by jfontan** [comment]
The last change moves the `mode` of `uast_mode` to the first parameter. If we want to make `lang` optional in the future we can do it without breaking compatibility.

**Pull request merged**